src/Twig/StimulusTwigExtension.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony WebpackEncoreBundle package.
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Twig;
+
+use Symfony\WebpackEncoreBundle\Exception\MissingPackageException;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class StimulusTwigExtension extends AbstractExtension
+{
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('stimulus_controller', [$this, 'renderStimulusController'], ['needs_environment' => true, 'is_safe' => ['all']]),
+        ];
+    }
+
+    public function renderStimulusController(Environment $env, array $data): string
+    {
+        if (!$data) {
+            return '';
+        }
+
+        $controllers = [];
+        $values = [];
+
+        foreach ($data as $controllerName => $controllerValues) {
+            $controllers[] = $controllerName;
+
+            foreach ($controllerValues as $key => $value) {
+                if (!is_scalar($value)) {
```

Comment:
why not scalars? shouldn't we always pass a json?

Reply:
Not according to Stimulus: https://stimulus.hotwire.dev/reference/values#types

Reply:
Actually json_encode doesn't: true needs to be encoded in 1, false in 0.

Reply:
OK, the controller decides how it is going to parse the data!
Thanks for the link.

Reply:
Yes and that's to usefulness of the Values API: the parsing is done by Stimulus :)